python/mxnet/gluon/trainer.py

```diff
@@ -259,6 +259,13 @@ def learning_rate(self):
         else:
             return self._optimizer.learning_rate
 
+    @property
+    def optimizer(self):
+        if isinstance(self._optimizer, opt.Optimizer):
+            return self._optimizer
+        else:
+            raise UserWarning("Optimizer has not been initialized yet")
```

Comment:
What if the user sets a custom optimizer here?

Reply:
a custom optimizer should still inherit the base Optimizer class. Gluon trainer does the check: https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/gluon/trainer.py#L123